packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Safe
+import { GnosisSafe as Safe } from "safe-contracts/GnosisSafe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { OwnerManager } from "safe-contracts/base/OwnerManager.sol";
+
+// Interfaces
+import { ILivenessModule2 } from "interfaces/safe/ILivenessModule2.sol";
+
+/// @title LivenessModule2
+/// @notice This module allows for challenge-based ownership transfer to a fallback owner
+///         when the Safe becomes unresponsive. The fallback owner can initiate a challenge,
+///         and if the Safe doesn't respond within the challenge period, ownership transfers
+///         to the fallback owner.
+/// @dev This is a singleton contract. To use it:
+///      1. The Safe must first enable this module using ModuleManager.enableModule()
+///      2. The Safe must then configure the module by calling enableModule() with parameters
+contract LivenessModule2 is ILivenessModule2 {
+    /// @notice Reserved address used as the previous owner to the first owner in a Safe
+    address public constant SENTINEL_OWNER = address(0x1);
+
+    /// @notice Mapping from Safe address to its configuration
+    mapping(address => ModuleConfig) public safeConfigs;
+
+    /// @notice Mapping from Safe address to active challenge start time (0 if no challenge)
+    mapping(address => uint256) public challengeStartTime;
+
+    /// @notice Semantic version.
+    /// @custom:semver 2.0.0
+    string public constant version = "2.0.0";
+
+    /// @notice Configures the module for a Safe that has already enabled it
+    /// @dev MUST only be callable by a Safe that has enabled this module
+    /// @param _config The configuration parameters for the module
+    function configure(ModuleConfig memory _config) external {
+        if (_config.livenessResponsePeriod == 0 || _config.fallbackOwner == address(0)) {
```

Comment:
Worth adding a comment explaining why we don't want to allow these conditions. Also, could maybe be two separate if statements with different errors for clarity.